python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/asset_decorator.py

```diff
@@ -0,0 +1,114 @@
+    AssetSpec,
+    BackfillPolicy,
+    PartitionsDefinition,
+    multi_asset,
+)
+from dagster._utils.security import non_secure_md5_hash_str
+
+from dagster_embedded_elt.sling.dagster_sling_translator import DagsterSlingTranslator
+from dagster_embedded_elt.sling.sling_replication import SlingReplicationParam, validate_replication
+
+
+def get_streams_from_replication(
+    replication_config: Mapping[str, Any],
+) -> Iterable[Mapping[str, Any]]:
+    """Returns a list of streams and their configs from a Sling replication config."""
+    return [
+        {"name": stream, "config": config}
+        for stream, config in replication_config.get("streams", {}).items()
+    ]
+
+
+def sling_assets(
+    *,
+    replication_config: SlingReplicationParam,
+    dagster_sling_translator: DagsterSlingTranslator = DagsterSlingTranslator(),
+    partitions_def: Optional[PartitionsDefinition] = None,
+    backfill_policy: Optional[BackfillPolicy] = None,
+    op_tags: Optional[Mapping[str, Any]] = None,
+) -> Callable[..., AssetsDefinition]:
+    """Create a definition for how to materialize a Sling replication stream as a Dagster Asset, as
+    described by a Sling replication config. This will create on Asset for every Sling target stream.
+
+    A Sling Replication config is a configuration that maps sources to destinations. For the full
+    spec and descriptions, see `Sling's Documentation <https://docs.slingdata.io/sling-cli/run/configuration>`_.
+
+    Args:
+        replication_config: Union[Mapping[str, Any], str, Path]: A path to a Sling replication config, or a dictionary of a replication config.
+        dagster_sling_translator: DagsterSlingTranslator: Allows customization of how to map a Sling stream to a Dagster AssetKey.
+        partitions_def: Optional[PartitionsDefinition]: The partitions definition for this asset.
+        backfill_policy: Optional[BackfillPolicy]: The backfill policy for this asset.
+        op_tags: Optional[Mapping[str, Any]]: The tags for this asset.
+
+    Examples:
+        Running a sync by providing a path to a Sling Replication config:
+
+        .. code-block:: python
+
+            from dagster_embedded_elt.sling import sling_assets, SlingResource, SlingConnectionResource
+
+            sling_resource = SlingResource(
+                connections=[
+                    SlingConnectionResource(
+                        name="MY_POSTGRES", type="postgres", connection_string=EnvVar("POSTGRES_URL")
+                    ),
+                    SlingConnectionResource(
+                        name="MY_DUCKDB",
+                        type="duckdb",
+                        connection_string="duckdb:///var/tmp/duckdb.db",
+                    ),
+                ]
+            )
+
+            config_path = "/path/to/replication.yaml"
+            @sling_assets(replication_config=config_path)
+            def my_assets(context, sling: SlingResource):
+                for lines in sling.replicate(
```

Comment:
I'm not seeing this `replicate` method in any of the tests or on the `SlingResource`. Should it be?

Reply:
yes :( this was the first PR in a stack but ran into Graphite woes, so now am trying to fix those before bringing back in the tests.